### ISSUE TITLE
ZeroDivisionError:float division by zero

### ISSUE THREAD

**Issue by HrishiDhondge** [openmm/openmm#3446]
I am getting this error while using pdbfixer, but when I trace back the error; this actually arises from the pdbstructure.py file.
To replicate this error: 
- 1) Download any NMR structure from PDB (2RS2)
- 2) Separate the chains using VMD as follows 
```
vmd -dispdev text 2RS2.pdb
set sel1 [atomselect top "chain A and protein"]
$sel1 writepdb protA.pdb
exit
```
- 3) Now use pdbfixer on the protA.pdb with all default parameters

This error arises due to the fact that VMD assigns the structure wrong/faulty unit cell parameters. For structures other than X-ray crystallography, it should be as follows (according to PDB file format documentation)
`CRYST1    1.000    1.000    1.000  90.00  90.00  90.00 P 1           1 `

But the separated chain from VMD has 
`CRYST1    0.000    0.000    0.000  90.00  90.00  90.00 P 1           1 `

and the main cause of the problem here. 
So I propose to add a little modification in a few lines from the pdbstructure.py script -
The original lines are as follows:
```
elif command == "CRYST1": 
    a_length = float(pdb_line[6:15])*0.1 
    b_length = float(pdb_line[15:24])*0.1 
    c_length = float(pdb_line[24:33])*0.1  
``` 

I propose one of the two ways:    
```
elif command == "CRYST1": 
    a_length = float(pdb_line[6:15]) *0.1
    b_length = float(pdb_line[15:24])*0.1
    c_length = float(pdb_line[24:33]) *0.1
    if (a_length ==b_length) and (b_length == c_length) and (c_length == 0):
        a_length, b_length, c_length = 1.0, 1.0, 1.0
``` 
The above method sets the default unit cell parameters for other methods, in case the parameters are set wrong in the PDB file.

```
elif command == "CRYST1": 
    a_length = float(pdb_line[6:15]) *0.1
    b_length = float(pdb_line[15:24])*0.1
    c_length = float(pdb_line[24:33]) *0.1
    if (a_length ==b_length) and (b_length == c_length) and (c_length == 0):
        continue
```
The above method simply ignores the unit cell parameters and proceed with the other computations.

Which one of these two would be better to incorporate in order to avoid such errors?

**Comment by peastman** [comment]
It's a good question.  I think I would go with this

```python
if not (a_length == b_length == c_length == 0):
    self._periodic_box_vectors = computePeriodicBoxVectors(a_length, b_length, c_length, alpha, beta, gamma)
```

Just ignore the whole CRYST1 record, since it contains no meaningful information and doesn't actually describe something that can be interpreted as a unit cell.  I avoided using `continue` since that creates a risk of bugs in the future, if someone should add more code inside the loop after the if/else block.

**Comment by HrishiDhondge** [comment]
I agree there might be a risk of bugs in the future, and to tackle that risk I think my first option will work better.
```
if (a_length == b_length == c_length == 0):
        a_length, b_length, c_length = 1.0, 1.0, 1.0
```

If we think about it, no unit cell will have the length (a = 0, b = 0, c = 0). And in the case of NMR structures, where we don't have the unit cell; the PDB format suggests that it should be (a = 1, b = 1, c = 1).

**Comment by HrishiDhondge** [comment]
I created a pull request by adding the mentioned code. This issue can be closed now. Thanks!

**Comment by peastman** [comment]
Fixed by #3469.
